### PR TITLE
Fix bypassing vaulting clumsy check with verb action.

### LINKED ISSUE #17423: Clumsy climbing bonks are borked
## Description

Climbing a table as a clown always fails if you drag yourself onto it and never fails if you alt + click. It looks like if the drag drop passes the random check, it just silently fails to put you on the table.

### LINKED ISSUE #24951: Using the vault verb action as someone with clumsy lets you vault anyway without bonking you
## Description

**Reproduction**

1. Choose a monkey ghost role
2. Try to vault over a table with your mouse
3. You get bonked
4. Now do it with right click vault
5. You can vault now without bonking

**Screenshots**

**Additional context**

### LINKED ISSUE #25951: Climbing `CanVault` check is nonsense and can be bypassed
## Description

In the `ClimbSystem`:
1. `CanVault` requirements are nonsense.
2. Using Vault verb bypasses `CanVault`.

Point 2. is self-explanatory.
**Point 1. needs a decision.** Currently the user needs a leg or (!) a foot in order to be able to vault which IMO is a total nonsense.
Vaulting is just renamed climbing (climbing when user is initiating the climb by themselves), and doesn't really have anything to do with legs or feet. Some creatures without legs or feet are able to climb (Slimes), and also some creatures with legs or feet cannot climb (Cockroach). Whether a creature can initiate climb on its own (vault) should be separate from having bodyparts[^1].

IMO it should be a separate "can climb on its own" bool data field in the `ClimbingComponent` and thus defined per prototype. Checking for foot or leg doesn't make sense.

### **Please preferably do not make a PR to fix point 2. before point 1. is fixed. Fixing both at once is fine.**
Fixing point 2. alone is easy but it also makes everything that hasn't got a foot or leg unable to vault, which surprisingly means disabling a lot of gameplay (looking at you, ghost roles).

[^1]: In the future - if anyone ever implements dismemberment - things like requiring body parts in order to climb could be supported, but that still should be a per-prototype setting. I.e. a humanoid can't climb without feet, but a slime can.

### PR DESCRIPTION
**Pull request by Tayrtahn**
## About the PR

Stops clumsy characters being able to climb/vault onto tables and such by using alt-click or the verb menu instead of drag-dropping.

This also fixes a separate bug where, when a clumsy character managed to pass the bonk check, the action would just cancel and the character didn't actually climb the object. Passing the check will now let the character climb.

Additionally, this fixes a oddity where the ability for mobs to climb was determined by whether or not they had feet/legs, which was arbitrary and problematic for creatures like slimes (which should be able to ooze up onto a table). This check could also be bypassed by using the verb instead of drag-dropping, so that was fixed too. The ability for mobs to climb is now controlled in the prototypes.

The default chance of a clumsy character bonking on a climbable has also been reduced from 75% to 50% to compensate somewhat.

## Why / Balance

Fixes #24951
Fixes #17423
Fixes #25951

## Technical details

ClimbSystem now raises AttemptClimbEvent as part of TryClimb, and aborts climbing if it gets cancelled. This makes sure that all code paths go through a clumsy check. BonkSystem now listens for AttemptClimbEvents and responds to them instead of trying to intercept DragDropEvents.
CanVault now gets checked in TryClimb as well, to prevent bypassing it.

The logic for climbing capabilities is now:
- The presence of ClimbingComponent makes an entity able to be placed on surfaces like tables.
- ClimbingComponent.CanClimb controls whether it can climb onto surfaces by drag-drop or verb.

The new field defaults to true to minimize changes to existing behavior. Some mobs will have gained the ability to climb when previously they couldn't, but that should be less impactful than the opposite and can be resolved by YML changes in the future.

## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- fix: Clumsy characters can no longer avoid bonking their heads by using the Climb verb.
- tweak: Clumsy characters are less likely to bonk their heads when trying to climb.

### CODE REVIEW

**Comment by themias** [comment]
I'd imagine this would also fix #17423.

**Comment by Tayrtahn** [comment]
Yeah, looks like it does!

**Comment by EmoGarbage404** [comment]
i would rather the bonkable system itself just get fixed considering that it's not very large and this hack is quite nasty.

**Comment by Tayrtahn** [comment]
Yeah, fair enough. Maybe I'll take a crack at that soon

**Comment by themias** [comment]
To prevent this from being debilitating, we could reduce BonkClumsyChance. (It's currently at 75% !) Dropping it down to, say, 5% would allow clowns to climb most of the time, with the occasional funny bonk.

**Comment by Tayrtahn** [comment]
Yeah, I'm not fond of relying on bugs for balance. Better to fix the bug and implement the desired balance properly.

**Comment by Tayrtahn** [comment]
It works certainly be possible to make ClumsyComponent have its own likelihood modifier too. So clowns could have a small chance of bonking while monkeys are more likely to bonk. Could even apply a boost when the character is drunk...

**Comment by Tayrtahn** [comment]
Lowered the default bonk chance to 50% and made the chance configurable as part of ClumsyComponent, so different entities can be given different levels of clumsiness.

File: Content.Shared/Interaction/Components/ClumsyComponent.cs [hunk 1]
```diff
@@ -9,14 +9,21 @@ namespace Content.Shared.Interaction.Components
...
         [ViewVariables(VVAccess.ReadWrite)]
         public DamageSpecifier ClumsyDamage = default!;
 
+        /// <summary>
+        /// Chance between 0 and 1 that the character will bonk when
+        /// trying to climb something.
+        /// </summary>
+        [DataField]
+        public float BonkChance = 0.5f;
+
```
Comment (metalgearsloth) [inline Content.Shared/Interaction/Components/ClumsyComponent.cs:22]: Moving 1 bonk field out is pretty weird, would be better to keep it all on the same component.

Comment (Tayrtahn) [reply Content.Shared/Interaction/Components/ClumsyComponent.cs:22]: I can put it back, I suppose. Would it be alright to add a ClumsyLevel property to ClumsyComponent and like multiply ClumsyComponent.ClumsyLevel * BonkableComponent.BonkClumsyChance to input into the clumsy roll instead?

I like the idea of clumsiness not being a hard binary, so some creatures can be just a little bit clumsy.

Comment (Tayrtahn) [reply Content.Shared/Interaction/Components/ClumsyComponent.cs]: On second thought, I'll save that for another PR.

**Comment by deltanedas** [comment]
> apply a boost when character is drunk

give every climbing mob clumsy set to 0% base chance then when they get drunk increase so disgruntled bar patrons trying to jump the counter slam their face :trollface:

**Comment by Tayrtahn** [comment]
Added a fix for CanVault only being checked with drag-drop interaction. Now the verb can't be used to bypass it.

Also removed the weird check for a leg or foot as part of CanVault. CanVault now checks if the user's ClimbingComponent has CanClimb set to true, or CanForceClimb set to true if dragging another entity onto something.

As discussed in #25951, this may make some things able to climb that couldn't before. This shouldn't be too big of a problem and can be fixed in a future PR.

**Comment by Tayrtahn** [comment]
No need to apologize! This is what PR reviews are for, after all.

I think I'll take the simpler route and get rid of CanClimbForce for now.

File: Content.Shared/Climbing/Events/AttemptClimbEvent.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    public EntityUid Climber;
+    public EntityUid Climbable;
+
+    public AttemptClimbEvent(EntityUid user, EntityUid climber, EntityUid climbable)
+    {
+        User = user;
+        Climber = climber;
+        Climbable = climbable;
+    }
+}
```
Comment (metalgearsloth) [inline Content.Shared/Climbing/Events/AttemptClimbEvent.cs:15]: record struct byref plz, rest looks good.

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**